src/Core/IThreadHandling.cs

```diff
@@ -53,6 +53,14 @@ public interface IThreadHandling
         /// </summary>
         Task RunOnUIThread(Action op);
 
+        /// <summary>
+        /// Executes the operation synchronously on the main thread.
+        /// If the caller is on the main thread already then the operation is executed directly.
+        /// If the caller is not on the main thread then the method will switch to the main thread,
+        /// then resume on the caller's thread when then the operation completes.
+        /// </summary>
+        void RunOnUIThreadSync(Action op);
```

Comment:
I added this as I think it makes more sense to have the run sync method and call it in the places that I had previously added "Forget" as these methods depend on it not being called async.